File: 1-easy/problems/1909-remove-one-element-to-make-the-array-strictly-increasing/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def canBeIncreasing(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        deleted = False
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                continue
            if deleted:
                return False
            deleted = True
            if i >= 2 and nums[i-2] > nums[i]:  # delete nums[i] or nums[i-1]
                nums[i] = nums[i-1]
        return True

File: 1-easy/problems/1909-remove-one-element-to-make-the-array-strictly-increasing/solution.ts
function canBeIncreasing(nums: any): boolean | number | string | any {
  """
  :type nums: List[int]
  :rtype: bool
  """
  deleted = false
  for (i in Array.from({length: 1, (nums}, (_, i) => i).length)) {
      if (nums[i] > nums[i-1]) {
          continue
      if (deleted) {
          return false;
      deleted = true
      if i >= 2 && nums[i-2] > nums[i]:  # delete nums[i] || nums[i-1]
          nums[i] = nums[i-1]
  return true;
}

export { canBeIncreasing };
